### ISSUE TITLE
Notes for the presentation to Maker DAO 

### ISSUE THREAD

**Issue by sergfeldman** [ubiquity/research#108]
## Intro

DAI is a stablecoin from Maker DAO and an abbreviation of Development AI - an AI-powered or AI-assisted development.

DAI services can be used in each phase of the software development lifecycle.

Agile SDLC for reference 
https://www.cleveroad.com/blog/agile-sdlc/ 
https://relevant.software/blog/agile-software-development-lifecycle-phases-explained/ 

### Services that Ubiquity can create in collaboration with Maker DAO

We can use 2 approaches for the DAI services - AI on the background and Copilot.

The Copilot concept is promoted by GitHub and Microsoft 
https://blogs.microsoft.com/blog/2023/09/21/announcing-microsoft-copilot-your-everyday-ai-companion/

# DAI services for organizations

**AI-powered repository manager** 

Covers the major software development steps on GitHub
- Create GitHub issues (details are in the issue below)
- https://github.com/ubiquity/research/issues/104 
- Review PRs on GitHub

**AI-powered team manager**

Covers the major people management steps on GitHub 
- Recruiting and adaptation (task-developer matchmaking)
- Motivation and incentives (Ubiquibot)
- Performance review (Ubiquibot)
- Education

# DAI service for individuals 

**AI-powered personal GitHub assistant**

Covers major individual activities on GitHub
- A personal AI assistant and a digital twin on GitHub (details we discussed in the issue below)
- https://github.com/ubiquity/github-agents/issues/1

@pavlovcik You can use these notes for the presentation to Maker DAO :)
@Draeieg @Keyrxng I would appreciate your feedback as well

**Comment by Keyrxng** [comment]
Can you provide more context on the presentation please @sergfeldman I think that'll be useful for me in adding some value here if I can

You've touched all the main bases I think but I'm unsure of how to contribute something meaningful on top of what you have already.

> The Copilot concept is promoted by GitHub and Microsoft

This is really interesting, a little controversial if you ask me. It sees everything from work, to browsing? But I agree that everyone is going to have a hyper personalized AI assistant in the next year or two in the palm of their hand.

**Comment by 0x4007** [comment]
There isn't all that much context at the moment but the gist:

I have an opportunity to pitch the founder of MakerDAO through direct introductions inside of maker to use our "DAO OS" (UbiquiBot)

They believe that rune will respond best to extremely visionary pitch. Perhaps with less of an emphasis on being pragmatic to solve problems today but more like change the world of tomorrow. 

The spotlight of the pitch is on UbiquiBot and the way that we run Ubiquity DAO

**Comment by Keyrxng** [comment]
Outstanding

Right okay I'll stew on it and see if I can add something of substance, is there an ETA on the meet?

**Comment by 0x4007** [comment]
Asap but realistically a couple weeks

**Comment by sergfeldman** [comment]
@pavlovcik Inspiration for an extremely visionary pitch 
https://youtu.be/gMsQO5u7-NQ

https://techcrunch.com/2023/09/30/humanes-ai-pin-debuts-on-the-paris-runway/

https://hu.ma.ne/

**Comment by Keyrxng** [comment]
> https://techcrunch.com/2023/10/03/induced-ai/
> 
> TLDR: Robotic Process Automation leveraging Zapier to automate workflows in the browser. Backed by Sam Altman (OpenAI CEO) among others, raised $2.3m in seed funding.

So where they are applying these RPA techniques for business processes, we are doing much the same but for development related processes.

We could apply this level of thinking to things like:

- automating hosting setup (and more?)
- automating project setup/configuration
- containerizing before or after
- etc and so forth and what not

The visionary pitch is:

UbiquiBot will serve many masters and they all each will have their own setups, containers, dbs etc and that increases time-till-contribution or in some cases may outright cause someone to stop trying if they encounter too many errors or it proves too difficult. Eliminating the need for any sort of project setup would be ridiculous in every possible way, hosting and domain debugging is an after thought but is still part of the development experience and with some brief looking around it seems to be a pretty untapped market.

Taking InducedAI for example they are just running headless chrome and taking input from the user in order to execute so why can't we do the same for dns setup and debugging, creating and managing clusters, whatever else it may be.

We can learn from their approach and apply it to any level of the developer experience, from setup to shipping. GPT-Engineer takes care of bootstrap to build but it won't setup your domain, help with DNS, collect or act upon analytics but it could.

This targets both individuals and organizations:

- Not every individual has that breadth of knowledge that would cover all aspects of development and some things can take days even weeks to learn and figure out how to setup or debug when you know how.

- Organisations benefit from this too, onboarding new starters again falls into the same space as the point above. They could also benefit from it from an optimization standpoint if they are already fully setup so-to-speak through the analytics side of things but also with any new module or products being built.

- Imagine a new module of something bigger which is also a standalone tool, something with the above abilities would ensure that everything is setup/containered/properly hosted in order for it to be accessible and functional as it is intended. I'm thinking along the lines of VPNs, tunnelling anything of that domain in terms of access or functionality.

> Imposter syndrome setting in reading this back lmao. Am I in the right area with my thinking at least? I can't tell if I'm still being too short sighted with problems of today than problems of tomorrow but It certainly fits into the DAI genre

**Comment by Keyrxng** [comment]
See reading the initial post @sergfeldman I feel that the idea is to keep things within the confines of GitHub but then reading your linked information about personal assistants and things I broadened my thinking

steer me right if I'm off on a tangent

**Comment by 0x4007** [comment]
I think for now the track to focus on is a DAO manager, currently with its nexus inside of GitHub (UbiquiBot) but with reach across all DAO channels (Twitter, Telegram, Discord, Website etc) 

Or slightly further future vision: DAOs being maintained by AIs with specific focuses. People can build their own DAOs by selecting and tweaking their AIs from some type of marketplace to get things running.

**Comment by sergfeldman** [comment]
@pavlovcik Arthur Hayes
> I will describe the craze I expect to see around new AI technology, which will be funded in large part with this toilet paper money

https://cryptohayes.substack.com/p/double-happiness

**Comment by sergfeldman** [comment]
@Keyrxng Thank you so much for the example of an RPA startup and a visionary pitch!
Regarding the project vision, it is worth focusing on the input from @pavlovcik

---

I was thinking about the "further future vision: DAOs being maintained by AIs with specific focuses".
When we talk about a focus for a person, we often mean his profession.
But LLMs like ChatGPT can obtain a new profession relatively quickly with the help of prompt engineering and relevant data.

Then what specific focuses should AI have for DAOs? :)
Focuses might be according to the functions, as described above for DAI - functions for the development process itself, and functions for managing people.

Please share if you have other ideas.

**Comment by Keyrxng** [comment]
I saw this tweet yesterday:

"Half a million $GTC was sent to the contract address and lost. 

The proposal was live for 10 days, passed the Tenderly simulation and was approved by 136 addresses. Nobody spotted it. 

DAOs need better tooling."

**Comment by Keyrxng** [comment]
I assume that it's a one-OS-fits-all, so where one DAO serves a purpose of say Web3 Security and another for Web3 development (Spearbit & DevDAO) they have different metrics to monitor their performance and efficiency in their core purpose while they have shared metrics like people management etc but we'd need to be able to accommodate most any goal or purpose.

For me AI breaks down into a few subcategories that could be applied:

- Automation
- Analytics
- Generation

Things we could possibly AI analyse:

- Core Purpose (agent based w/ user input to accommodate arbitrary metrics and purposes)
- Shared Attributes (onboarding, "HR" or people management, offboarding, transaction proofing re:GTC)
- Social Engagement/Sentiment
- Internal Engagement/Sentiment

Things we could possibly AI automate:

> I don't think there is enough faith to automate anything with monetary value such as payments etc

- Accountancy (tx recording, visualization etc)
- Transaction safety (from a ML standpoint, it would learn from accountancy from previous tx history then for any new txs they'd follow the same validation as normal then last line of defence would be AI automated 'tenderly' but better)
- Core purpose assistance (Kit an agent with common tooling, web indexing, Slack, GitHub)

Things we could possibly generate:

- It's likely covered in repo manager but anything docs related despite tools already existing for each, to embed them all into one agent would be lethal. Codebase docs, Onboarding docs, 'AI whitepaper', storybooks etc
- Marketing campaigns / Social media / Blog content
- Accountancy belongs here too as there would be a lot of generation going in specifically for that.


I'm thinking very organisationally here...

**Comment by sergfeldman** [comment]
@Keyrxng Great input. The most striking me points:

>they have shared metrics like people management etc but we'd need to be able to accommodate most any goal or purpose.

I think the same way.

>Things we could possibly generate

On GitHub the core entities are issues, PRs, discussions.
There are several code generating and PR review services on the market, but issue writing is not developed at the moment.

**Comment by sergfeldman** [comment]
[State of AI report 2023 ](https://drive.google.com/file/d/1lwIFS_N39v0830Q3l3fz7bYfZvp4LsNL/view?usp=drive_link)

Research

- GPT-4 lands and demonstrates a capabilities chasm between proprietary and next-best open source alternatives, while also validating the power of
reinforcement learning from human feedback.
- Efforts grow to clone or beat proprietary model performance with smaller models, better datasets, longer context...powered by LLaMa-1/2.
- It’s unclear how long human-generated data can sustain AI scaling trends (some estimate that data will be exhausted by LLMs by 2025) and what the effects
of adding synthetic data are. Videos and data locked up in enterprises are likely up next.
- LLMs and diffusion models continue to offer gifts to the life science community by producing new breakthroughs for molecular biology and drug discovery.
- Multimodality becomes the new frontier and excitement around agents of all flavors grows substantially.

Industry

- NVIDIA rips into the $1T market cap club with voracious demand for its GPUs from nation states, startups, big tech and researchers alike.
- Export controls rate limit advanced chip sales to China, but major chip vendors create export control-proof alternatives.
- Led by ChatGPT, GenAI apps have a breakout year across image, video, coding, voice or CoPilots for everyone, driving $18B of VC and corporate investments.

Politics

- The world has divided into clear regulatory camps, but progress on global governance remains slower. The largest AI labs are stepping in to fill the vacuum.
- The chip wars continue unabated, with the US mobilising its allies, and the Chinese response remaining patchy.
- AI is forecast to affect a series of sensitive areas, including elections and employment, but we’re yet to see a significant effect.

Safety

- The existential risk debate has reached the mainstream for the first time and intensified significantly.
- Many high-performing models are easy to ‘jailbreak’. To remedy RLHF challenges, researchers are exploring alternatives, e.g. self-alignment and pretraining
with human preferences.
- As capabilities advance, it’s becoming increasingly hard to evaluate SOTA models consistently. Vibes won’t suffice.

**Comment by sergfeldman** [comment]
This example can be used to show the Copilot for a bounty hunter
https://www.careerflow.ai/#more-info

**Comment by sergfeldman** [comment]
Scale your team with autonomous AI Agents
Automate manual workflows with AI Agents to boost productivity and let you and your team focus on the work that matters most.
https://www.beam.ai/

**Comment by sergfeldman** [comment]
New AI-powered organisation 

![IMG_20231127_180605_970](https://github.com/ubiquity/research/assets/15567379/7be7f9f5-3c8d-49f2-a9cb-c9bf26d0384a)

**Comment by sergfeldman** [comment]
Thoughts on the AI-powered SDLC
https://medium.com/@rama.sathish/ai-powered-software-development-life-cycle-1ac599ad38bb

**Comment by sergfeldman** [comment]
Maker DAO Endgame documentation
https://endgame.makerdao.com/endgame/overview overview
https://endgame.makerdao.com/subdaos/overview SubDAOs
https://endgame.makerdao.com/tokenomics/subdao-tokenomics SubDAOs tokenomics
https://mips.makerdao.com/mips/details/MIP106#3-3-1-2 use of AI
